### DISCUSSION TITLE
cannot repeat the RMSE of water dataset

### DISCUSSION THREAD

**Discussion by phys-chem** [ACEsuit/mace#346] [Q&A] [answered]
Hello! I also meet this problem. I download the original dataset https://github.com/BingqingCheng/ab-initio-thermodynamics-of-water/tree/master/training-set, and convert units by myself.
I am repeating the results of liquid water in J. Chem. Phys. 159, 044118 (2023). I only get a RMSE of 4meV/H2O. Here is my script:

mace_run_train
--name="water1593"
--train_file="../../water1593_train.xyz"
--valid_file="../../water1593_test.xyz"
--E0s="average"
--model="MACE"
--num_interactions=2
--num_channels=64
--max_L=0
--correlation=3
--r_max=6.0
--forces_weight=1000
--energy_weight=10
--batch_size=2
--valid_batch_size=4
--max_num_epochs=2000
--start_swa=400
--scheduler_patience=15
--patience=30
--eval_interval=4
--ema
--swa
--error_table='PerAtomRMSE'
--default_dtype="float32"
--device=cuda
--seed=123
--restart_latest
--save_cpu

The main difference is that I use float32. Could it be the reason of higher energy RMSE? Thank you very much!
[train.log](https://github.com/ACEsuit/mace/files/14614727/train.log)

**Comment by ilyes319** [comment, marked as answer]
Hey, please use float64. Also the default swa changed, here is an equivalent input:

```
python <mace_repo_dir>/mace/cli/run_train.py \
    --name="water_1k_small" \
    --train_file="train.xyz" \
    --valid_fraction=0.05 \
    --E0s="average" \
    --model="MACE" \
    --num_interactions=2 \
    --num_channels=64 \
    --max_L=0 \
    --correlation=3 \
    --r_max=6.0 \
    --forces_weight=1000 \
    --energy_weight=10 \
    --swa_forces_weight=10 \
    --swa_energy_weight=1000 \
    --energy_key="TotEnergy" \
    --forces_key="force" \
    --batch_size=2 \
    --valid_batch_size=4 \
    --max_num_epochs=800 \
    --start_swa=400 \
    --scheduler_patience=15 \
    --patience=30 \
    --eval_interval=4 \
    --ema \
    --swa \
    --error_table='PerAtomMAE' \
    --default_dtype="float64"\
    --device=cuda \
    --seed=123 \
    --restart_latest \
    --save_cpu \
```

**Reply by phys-chem** [reply]
Thank you! The default "swa" weight changed, right? 
In addition, 95% data are used for training , 5% data are used for validation in the input. It seems that BPNN and REANN do not use so much data for training.
![image](https://github.com/ACEsuit/mace/assets/62685413/c921bb36-7c04-4de5-852e-5799626cd1af)

**Reply by ilyes319** [reply]
@davkovacs can help with that

**Reply by phys-chem** [reply]
Thank you! I can reproduce it with these parameters.

**Reply by phys-chem** [reply]
By the way, does float64 really improve the accuracy？I found there are no obvious improvement for the water data set. Do you have compare the difference between float32 and float64? Thank you very much!

**Reply by bernstei** [reply]
Minimizations (with reasonably tight tolerance, and especially stress/cell relaxation) tends to fail with float32.

**Reply by phys-chem** [reply]
Thank you for the reply! Therefore, it affects minimization. The fitting errors are close between float32 and float64. Is my understanding correct?

**Reply by ilyes319** [reply]
You will not see much difference between float64 and float32 in training accuracy. As Noam said, it is mostly when doing iterative methods, like geometry optimizations, that you will see differences. In my experience, NVT MD is fine with float32 but for geometry optimization, you should use float64.
